week4/week4-hw/src/components/Login/Login.tsx

```diff
@@ -0,0 +1,74 @@
+import Input from '../common/Input/Input';
+import Button from '../common/Button/Button';
+import { Link, useNavigate } from 'react-router-dom';
+import { loginArticle } from './Login.css';
+import { useMutation } from '@tanstack/react-query';
+import { apiRequest } from '../../fetching/axios';
+import { useState } from 'react';
+
+interface LoginRequestBody {
+    loginId: string;
+    password: string;
+}
+
+const Login = () => {
+    const [userId, setUserId] = useState('');
+    const [password, setPassword] = useState('');
+    const navigate = useNavigate();
+
+    const loginMutation = useMutation<unknown, Error, LoginRequestBody>({
+        mutationFn: async (requestBody) => {
+            const signinResponse = await apiRequest.post('/api/v1/auth/signin', requestBody);
+            localStorage.setItem('userId', signinResponse.data.data.userId);
+            return signinResponse;
+        },
+        onSuccess: async (data) => {
+            try {
+                const userInfoResponse = await apiRequest.get('/api/v1/users/me', {
+                    headers: { userId: data.data.data.userId },
+                });
+                localStorage.setItem('userName', userInfoResponse.data.data.nickname);
+                navigate('/mypage/info');
+            } catch (error) {
+                console.error('사용자 정보 가져오기 실패:', error);
+            }
+        },
+        onError: (error) => {
+            console.error('로그인 실패:', error);
+        },
+    });
+
+    const handleLogin = () => {
+        loginMutation.mutate({ loginId: userId, password: password });
+    };
```

Comment:
mutation의 다른 옵션을 사용하는게 아니라면

```suggestion
  const { mutate: loginMutation } = useMutation<unknown, Error, LoginRequestBody>

    const handleLogin = () => {
        loginMutation({ loginId: userId, password: password });
    };
```
이런식으로도 표현이 가능합니다